pkg/action/upgrade.go

```diff
@@ -122,12 +123,33 @@ func (u *Upgrade) prepareUpgrade(name string, chart *chart.Chart, vals map[strin
 		return nil, nil, errMissingChart
 	}
 
-	// finds the deployed release with the given name
-	currentRelease, err := u.cfg.Releases.Deployed(name)
+	// finds the last non-deleted release with the given name
+	lastRelease, err := u.cfg.Releases.Last(name)
 	if err != nil {
+		// to keep existing behavior of returning the "%q has no deployed releases" error when an existing release does not exist
+		if strings.Contains(err.Error(), "no revision for release") {
```

Comment:
Would be nice if this were more statically typed. Can `errors.Is` be used?

Reply:
The error is not a constant.  It's a string built with the name of the release in it (hence why I used strings.Contains.  Is there another method I don't know?

Reply:
My apologies, was not familiar with the codebase, it just looked incredibly.. Wrong..
But alas this seems to be the intended design throughout the codebase, so nothing wrong with it here.